clean_air/util/dataframe_converter.py

```diff
@@ -0,0 +1,148 @@
+    # It is as yet untested and unfinished; pulled from iris but in need of
+    # simplification and modification.
+    data = as_pandas_data(cube.data)
+    if not index:
+        if cube.dim_coords:
+            index = _as_pandas_coord(cube.dim_coords[0])
+    else:
+        index = _as_pandas_coord(index)
+
+    series = pd.Series(data, index)
+    if not copy:
+        _assert_shared(data, series)
+
+    return series
+
+
+def _make_geo(cube, x_coord, y_coord):
+    """
+    Helper function for conversion to geodataframe(s)
+    :param cube: cube for conversion
+    :param x_coord: iris dim coord representing x direction
+    :param y_coord: iris dim coord representing y direction
+    :return: list of geodataframes
+    """
+    dataframes = []
+    for sub_cube in cube.slices([x_coord, y_coord], ordered=True):
+        # Get the coordinates, data and geometry in the right
+        # order here:
+        x_coords_pd = []
+        y_coords_pd = []
+        data_pd = []
+        geom = []
+        i = 0
+        for x in x_coord.points:
+            # Reset the y-coord index to 0 each time we loop over the x-coord:
+            j = 0
+            for y in y_coord.points:
+                x_coords_pd.append(x)
+                y_coords_pd.append(y)
+                data_pd.append(sub_cube.data[i, j])
+                geom.append(Point(x, y))
+                # i and j represent the index values of the coords, so
+                # they need to stop counting at the end of the coord arrays.
+                if j < len(y_coord.points) - 1:
+                    j += 1
+            if i < len(x_coord.points) - 1:
+                i += 1
+
+        # Now construct pandas dataframe object for this sub_cube:
+        pandas_df = pd.DataFrame({'x_coord': x_coords_pd,
+                                  'y_coord': y_coords_pd,
+                                  'data': data_pd})
+
+        # Then make geometry in correct form (which is a list of
+        # shapely.geometry.Points objects) and convert to GeoDataFrame:
+        geo_df = GeoDataFrame(pandas_df, geometry=geom)
+        dataframes.append(geo_df)
+
+    return dataframes
+
+
+def convert_to_geodf(cube, restitch=True):
+    """
+    Callable for converting iris-style cubes into geopandas dataframes.
+
+    Note: This conversion is only possible for 2-dimensional cubes, so input
+    cubes with more than two dimensions will be sliced up before conversion
+    and then restitched if necessary.  Cubes with less than two dimensions
+    will be converted to a series instead.
+
+    :param cube: input cube.
+    :param restitch: Boolean representing whether to concatenate the
+    dataframes after conversion to pandas arrays.
+    :return: pandas-style dataframe(s) or series, depending on input (see note).
+    """
+    geodataframes = series = geodataframe = None
+    if len(cube.dim_coords) == 2:
+        restitch = False
```

Comment:
So, for 3+d cubes, we get a single dataframe but can explicitly ask for a list, but for 2d cubes we get a list containing exactly one dataframe and cannot ask for just the dataframe? This is very counterintuitive.

Reply:
It is a bit isn't it?  OK, I'll have another look at this and see if I can arrange it a bit better.

Reply:
OK so if I remove this condition then all 2D cubes will be sliced up over the xy coords just like 3+D cubes, but there don't seem to be any problems with that or the re-concatenation, so I'm gonna remove it.